src/System.Net.HttpListener/src/System/Net/Windows/RequestContextBase.cs

```diff
@@ -48,7 +50,14 @@ public void Dispose()
             Dispose(true);
         }
 
-        protected virtual void Dispose(bool disposing) { }
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_backingBuffer != IntPtr.Zero)
```

Comment:
Please review IDisposable pattern.  Not sure if you should add an 'if' check for "disposing == true" here.

Reply:
I think that the current behavior is correct, based on [this page](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#dispose-and-disposeboolean).

Reply:
ok.  What about a 'finalizer'/  Does this have one already and do we need one.  Reading the pattern description from your link above says this:

>You implement the IDisposable interface and an additional Dispose(Boolean) method, and you also override the Object.Finalize method. You must override Finalize to ensure that unmanaged resources are disposed of if your IDisposable.Dispose implementation is not called by a consumer of your type. If you use the recommended technique discussed in the previous bullet, the System.Runtime.InteropServices.SafeHandle class does this on your behalf.

Reply:
RequestContextBase does implement a finalizer that calls Dispose(bool), freeing the unmanaged resources it has acquired. I'm not very familiar with the pattern though, so I'll read through the rest of that dispose/finalize article and make sure I'm doing everything correctly.

Reply:
I've checked it out and am confident the dispose pattern is correctly implemented here.